### DISCUSSION TITLE
[v4] Underscores in arbitrary variant with class name are converted to spaces

### DISCUSSION THREAD

**Discussion by sntran** [tailwindlabs/tailwindcss#15277] [Help]
**What version of Tailwind CSS are you using?**

`4.0.0-beta.4`

**What build tool (or framework if it abstracts the build tool) are you using?**

`@tailwindcss/cli@4.0.0-beta.4`

**What version of Node.js are you using?**

v22.2

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/vIy4AY0qLN

**Describe your issue**

Tailwind replaces the double underscores in arbitrary variant with spaces, from `wp-block-cover__inner-container` to `wp-block-cover  inner-container`.

![image](https://github.com/user-attachments/assets/b0ebc26e-997f-4ec9-aa0d-3fc6df00452f)

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content#using-spaces-and-underscores):

> Since whitespace denotes the end of a class in HTML, replace any spaces in an arbitrary value with an underscore:
>
> ```html
> <div class="before:content-['Hello_World']">
>   
> </div>
> ```
> If you need to include an actual underscore, you can do this by escaping it with a backslash:
> 
> ```html
> <div class="before:content-['Hello\_World']">
>   
> </div>
> ```

So for your case: https://play.tailwindcss.com/Vxl7XFnGA7

**Comment by sntran** [comment]
I would vouch for Tailwind to check if the variant is for a class and don't do the replacement, instead of requiring the user to manually escape it. This BEM style is very popular in the WP world, so manually escaping them would be tedious.

I understand that one should not use these kind of variant much, but when dealing with WP, a lot of time we have to do one-off override like the above.
